File: uhrwerk-core/src/main/java/io/qimia/uhrwerk/config/representation/JDBC.java
package io.qimia.uhrwerk.config.representation;

import io.qimia.uhrwerk.config.ConfigException;

public class JDBC extends Representation{
    private String jdbc_url;
    private String jdbc_driver;
    private String user;
    private String pass;

    public JDBC(){}

    public String getJdbc_url() {
        return jdbc_url;
    }

    public void setJdbc_url(String jdbc_url) {
        this.jdbc_url = jdbc_url;
    }

    public String getJdbc_driver() {
        return jdbc_driver;
    }

    public void setJdbc_driver(String jdbc_driver) {
        this.jdbc_driver = jdbc_driver;
    }

    public String getUser() {
        return user;
    }

    public void setUser(String user) {
        this.user = user;
    }

    public String getPass() {
        return pass;
    }

    public void setPass(String pass) {
        this.pass = pass;
    }

    @Override
    public void validate(String path){
        path += "jdbc/";
        if(jdbc_url == null){
            throw new ConfigException("Missing field: " + path + "jdbc_url");
        }
        if(jdbc_driver == null){
            throw new ConfigException("Missing field: " + path + "jdbc_driver");
        }
        if(user == null){
            throw new ConfigException("Missing field: " + path + "user");
        }
        if(pass == null){
            throw new ConfigException("Missing field: " + path + "pass");
        }
    }
}

File: uhrwerk-config/src/main/kotlin/io/qimia/uhrwerk/config/representation/JDBC.kt
package io.qimia.uhrwerk.config.representation

import com.fasterxml.jackson.annotation.JsonProperty
import io.qimia.uhrwerk.config.builders.ConfigException

data class JDBC(
    override var name: String? = null,
    @JsonProperty("jdbc_url")
    var jdbcUrl: String? = null,
    @JsonProperty("jdbc_driver")
    var jdbcDriver: String? = null,
    var user: String? = null,
    var password: String? = null
) : Connection() {
    override fun validate(path: String) {
        var path = path
        //super.validate(path)
        path += "jdbc/"
        if (jdbcUrl == null) {
            throw ConfigException("Missing field: " + path + "jdbc_url")
        }
        if (user == null) {
            throw ConfigException("Missing field: " + path + "user")
        }
        if (password == null) {
            throw ConfigException("Missing field: " + path + "pass")
        }
    }
}